src/builder/QuarkBuilder.sol

```diff
@@ -377,19 +457,55 @@ contract QuarkBuilder {
         }
     }
 
+    function getWrapperCounterpartBalance(
+        string memory assetSymbol,
+        uint256 chainId,
+        Accounts.ChainAccounts[] memory chainAccountsList
+    ) internal pure returns (uint256) {
+        if (TokenWrapper.hasWrapperContract(chainId, assetSymbol)) {
+            // Add counterpart balance to balanceOnChain
+            return Accounts.getBalanceOnChain(
+                TokenWrapper.getWrapperCounterpartSymbol(chainId, assetSymbol), chainId, chainAccountsList
+            );
+        }
+
+        return 0;
```

Comment:
Ok.  But I found putting it return 0 if not found making some logics simpler. 
Simply a counterpart of 0 balance, and keep most logics the same to simplify the handling in building step.

Reply:
It looks like in every place other than 1 call site for this function, the token is guaranteed to have a counterpart because we check beforehand. If this function is called for a function with no counterpart, I think that is a bug in our implementation and it is good practice to revert.